demisto_sdk/commands/prepare_content/preparers/marketplace_suffix_preparer.py

```diff
@@ -26,37 +23,51 @@ def prepare(
         Returns: A (possibliy) modified content item data
 
         """
-        if supported_marketplaces is None:
-            supported_marketplaces = list(MarketplaceVersions)
-        if not (
-            suffix := MarketplaceSuffixPreparer.MARKETPLACE_TO_SUFFIX.get(
-                current_marketplace, ""
-            )
-        ):
-            return data
-        suffix_len = len(suffix)
+        suffix = f"{SEPARATOR}{current_marketplace.value}"
+        suffixes = [suffix]
+        if current_marketplace == MarketplaceVersions.XSOAR_ON_PREM:
+            suffixes.append(f"{SEPARATOR}{MarketplaceVersions.XSOAR.value}")
+        if current_marketplace == MarketplaceVersions.XSOAR_SAAS:
+            suffixes.append(f"{SEPARATOR}{MarketplaceVersions.XSOAR.value}")
 
         def fix_recursively(datum: Any) -> Any:
             if isinstance(datum, list):
                 return [fix_recursively(item) for item in datum]
 
             elif isinstance(datum, dict):
-                # performs the actual fix, without accessing the MARKETPLACE_TO_SUFFIX dictionary.
                 for key in tuple(
                     datum.keys()
                 ):  # deliberately not iterating over .items(), as the dict changes during iteration
                     value = datum[key]
-                    if isinstance(key, str) and key.casefold().endswith(suffix):
-                        clean_key = key[:-suffix_len]  # without suffix
+                    if isinstance(value, (list, dict)):
+                        fix_recursively(value)
+                        continue
+                    if SEPARATOR not in key:
+                        continue
+                    for suffix in suffixes:
+                        # iterate each suffix to see if it's relevant for the key.
+                        # the order of the suffixes matter!
```

Comment:
lets go over the cases, I like to understand the order importance.

Reply:
OK.
In general: if running on XSOAR_SAAS and we have the following keys:
<key>:xsoar_saas
<key>:xsoar
<key>

We should priorotize the xsoar_saas key over the xsoar. However, if there is only xsoar, we should take it and not the regular key.